src/frontend/src/optimizer/rule/project_agg_merge.rs

```diff
@@ -0,0 +1,52 @@
+// Copyright 2022 Singularity Data
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use super::super::plan_node::*;
+use super::{BoxedRule, Rule};
+
+/// Merge [`LogicalAgg`] <- [`LogicalProject`] to [`LogicalAgg`].
+pub struct ProjectAggMergeRule {}
```

Comment:
How about rename it to `AggProjectMergeRule`? Keep the rule name consistent with the match pattern.